### DISCUSSION TITLE
Welcome to Water-Physics-Overhaul Discussions!

### DISCUSSION THREAD

**Discussion by Felicis** [Felicis/Water-Physics-Overhaul#19] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by djmrFunnyMan** [comment]
Idea for a config setting: Packet Precision

RN water has 8 levels (packets) in this mod. Being able to lower this precision would help with performance.

Settings
Normal: 8 levels (current)
Decreased: 4 levels (2,4,6,8 packets)
Low: 2 levels (4,8 packets)
Lowest: 1 level (8 packets, only full blocks)

With lower precision the performance should improve because the more water would move at once (it wouldn't split up as much so less blocks updating) and it wouldn't have to equalize as often (or at all on the lowest setting)

The lowest setting would especially be useful if you want to let's say flood a big cave quickly because only entire water blocks would be moved at a time and no equalization would need happen.

**Reply by Felicis** [reply]
good idea. It will take some tweaking all over the place, especially how to handle upgrading a world from higher precision to lower precision, but it sounds totally feasible.

**Reply by djmrFunnyMan** [reply]
I have a few ideas how to handle that. Honestly none seem good enough for me.
1. Upon loading the world after decreasing accuracy round all the packets that now cannot exist up (this should happen on world loading instead of after, sort of like converting pre 1.18 worlds to that version). This would cause issues since a lot of water would be generated by hey at least you can load into the world and it's the simplest option.
2. The water stays as it was. But water blocks suck surrounding water blocks up to reach a possible level.
3. The water stays as it was. Then if the water is disturbed by the player (for example making a hole and letting it drain somewhere) it'll naturaly reach a possible packet.
4. Don't allow loading into a world after lowering precision. (meh)
5. The water stays as it was, but water with wrong packets cannot interact with water that's correct.

**Reply by Felicis** [reply]
Was just thinking about this now and option 3 seems the most reasonable and simple. Also I think Minecraft uses similar assumptions for saving and loading chunks :)

**Comment by NunQmad** [comment]
Hello. Thank you very much, for you job. I have some problem on version 1.18.2, water and lava disappear at altitude coordinates below zero. As I understand it, this is due to the fact that before 1.17 it started from 0, and after 1.17 from -64, I looked in the source code of the mod, but did not find anything similar. Could you help me fix this?
